web-bundle/src/main/java/com/graphhopper/resources/RouteResource.java

```diff
@@ -273,7 +273,14 @@ private void removeLegacyParameters(GHRequest request) {
             throw new IllegalArgumentException("Alternatives are currently not yet supported for GPX");
         }
 
-        long time = timeString != null ? Long.parseLong(timeString) : System.currentTimeMillis();
```

Comment:
This was the bug: For GET a missing `gpx.millis` would give null, but for POST it would be an empty String.